src/lcm/next_state.py

```diff
@@ -7,31 +7,48 @@
 
 """
 
+from collections.abc import Callable
+
 from dags import concatenate_functions
 from dags.signature import with_signature
+from jax import Array
 
 from lcm.functools import all_as_args
 from lcm.interfaces import InternalModel
 from lcm.random_choice import random_choice
+from lcm.typing import Scalar, Target
+
+
+def get_next_state_function(
+    model: InternalModel, target: Target
+) -> Callable[..., dict[str, Scalar]]:
+    """Get function that computes the next states of the model.
+
+    Args:
+        model: Internal model.
+        target: Target of the function.
+
+    Returns:
+        Function that computes the next states of the model.
 
+    """
+    if target == Target.SOLVE:
+        return _get_next_state_function_solution(model)
 
-def get_next_state_function(model: InternalModel, target):
-    if target == "solve":
-        out = _get_next_state_function_solution(model)
-    elif target == "simulate":
-        out = _get_next_state_function_simulation(model)
-    else:
-        raise ValueError(f"Target must be 'solution' or 'simulation'. Got {target}.")
+    if target == Target.SIMULATE:
+        return _get_next_state_function_simulation(model)
 
-    return out
+    raise ValueError(f"Invalid target: {target}")
 
 
 # ======================================================================================
 # Solution
 # ======================================================================================
 
 
-def _get_next_state_function_solution(model: InternalModel):
+def _get_next_state_function_solution(
```

Comment:
```suggestion
def _get_next_state_function_for_solution(
```
Else it is a bit ambiguous, could be the function's solution...